articles/connections/database/custom-db/create-db-connection.md

```diff
@@ -0,0 +1,152 @@
+
+```js
+function login(email, password, callback) {
+  var connection = mysql({
+    host: 'localhost',
+    user: 'me',
+    password: 'secret',
+    database: 'mydb'
+  });
+
+  connection.connect();
+
+  var query = "SELECT id, nickname, email, password " +
+    "FROM users WHERE email = ?";
+
+  connection.query(query, [email], function (err, results) {
+    if (err) return callback(err);
+    if (results.length === 0) return callback(new WrongUsernameOrPasswordError(email));
+    var user = results[0];
+
+    bcrypt.compare(password, user.password, function (err, isValid) {
+      if (err) {
+        callback(err);
+      } else if (!isValid) {
+        callback(new WrongUsernameOrPasswordError(email));
+      } else {
+        callback(null, {
+          id: user.id.toString(),
+          nickname: user.nickname,
+          email: user.email
+        });
+      }
+    });
+
+  });
+}
+```
+
+The above script connects to a MySQL database and executes a query to retrieve the first user with `email == user.email`.
+
+With the **bcrypt.compareSync** method, it then validates that the passwords match, and if successful, returns an object containing the user profile information including **id**, **nickname**, and **email**.
+
+This script assumes that you have a **users** table containing these columns. The **id** returned by Login script is used to construct the **user ID** attribute of the user profile. If you are using multiple custom database connections, then **id** value must be unique across all the custom database connections to avoid **user ID** collisions. Our recommendation is to prefix the value of **id** with the connection name (omitting any whitespace).
+
+Be sure to **Save** your changes. Note that clicking **Try** to test your script will also save your script.
+
+## Step 3: Add configuration parameters
+
+You can store parameters, like the credentials required to connect to your database, in the **Settings** section below the script editor. These will be available to all of your scripts, and you can access them using the global configuration object.
+
+You can access parameter values using the `configuration` object in your database action scripts (i.e. `configuration.MYSQL_PASSWORD`).
+
+![Custom database settings](/media/articles/connections/database/mysql/db-connection-configurate.png)
+
+Use the added parameters in your scripts to configure the connection. For example, you might add the following the MySQL Login template:
+
+```js
+function login (username, password, callback) {
+  var connection = mysql({
+    host     : configuration.MYSQL_HOST,
+    user     : 'me',
+    password : configuration.MYSQL_PASSWORD,
+    database : 'mydb'
+  });
+}
+```
+
+## Summary
+
+In this article, we showed you how to configure your database for use with Auth0 as an identity provider. You: 
+
+1. Created an Auth0 database connection
+2. Created database action scripts
+3. Added configuration parameters
```

Comment:
what's the next logical step? do we want a next-steps section here or are we removing this notion?